src/utils/telemetry.ts

```diff
@@ -0,0 +1,168 @@
+import os from 'os';
+import { version } from '../../package.json';
+import path from 'path';
+import fs from 'fs/promises';
+import crypto from 'crypto';
+import { logger } from './logger';
+
+export async function isFile(path: string): Promise<boolean> {
+  try {
+    const stat = await fs.stat(path);
+    return stat.isFile();
+  } catch {
+    return false;
+  }
+}
+
+/** @type {boolean} See envvar.md. */
+export const MY_PORTAL_ENABLED = !!process.env.MY_PORTAL && process.env.MY_PORTAL !== '0';
+
+export const STAGE_USER_NAME =
+  // Not every username is `first-last`, if `MY_PORTAL` looks like a username use that directly
+  MY_PORTAL_ENABLED && process.env.MY_PORTAL?.includes('-')
+    ? process.env.MY_PORTAL.toLowerCase()
+    : os.userInfo().username.replace(/\./g, '-');
+
+export const DEVVIT_PORTAL_URL = (() => {
+  if (MY_PORTAL_ENABLED) {
+    return `https://reddit-service-devvit-dev-portal.${STAGE_USER_NAME}.snoo.dev` as const;
+  }
+
+  return 'https://developers.reddit.com' as const;
+})();
+
+export const DEVVIT_PORTAL_API = `${DEVVIT_PORTAL_URL}/api` as const;
+
+export type HeaderTuple = readonly [key: string, val: string];
+
+export const HEADER_USER_AGENT = (): HeaderTuple => [
+  'user-agent',
+  `Devvit/MCP/${version} Node/${process.version.replace(/^v/, '')}`,
+];
+
+export const HEADER_DEVVIT_MCP = (): HeaderTuple => ['x-devvit-mcp', 'true'];
+
+export const HEADER_DEVVIT_CANARY = (val: string): HeaderTuple => {
```

Comment:
:no_bell: are we going to use these headers somewhere externally? if not, consider inlining. if so, consider extracting to a http.ts file.

Reply:
Copy pasta from the CLI package